lib/cram.ml

```diff
@@ -105,6 +105,17 @@ let of_lines t =
       (pad, aux cmd [] [] t)
   | `Command cmd :: t -> (pad, aux [ cmd ] [] [] t)
   | [] -> (0, [])
+  | `Output line :: _ ->
+      if String.length line > 0 && line.[0] = '$' then
+        failwith
+          "Blocks must start with a command or similar, not with an output \
+           line. To indicate a line as a command, start it with a dollar \
+           followed by a space."
+      else
+        failwith
+          "Blocks must start with a command or similar, not with an output \
+           line. Is there maybe a spare empty line somewhere, for example \
+           between the output and its input?"
```

Comment:
I'm not sure it's common to give hints to the user phrased as a question (some might expect to have to type an answer), how about we rephrase the last sentence as something like: `Please check there is no empty line between the command and its output.` ? (or something similar)

Reply:
In Rust, for example, a lot of error messages include a hint of what might have gone wrong phrased as "Did you mean ...?". But it's true: now that I think about it, in OCaml it doesn't seem to be common to have questions in error messages and therefore could lead to misunderstanding. Just as a side-note: also in Ocaml I've come across a few questions in error messages, but always phrased carefully (for example explicitly as a hint), e.g. when using `ppxlib`'s `metaquot` without providing a location in the scope, it says 
"Unbound value loc 
Hint: Did you mean log or lor?"

Anyways, I agree that it's better to rephrase it, so I've done that.